### ISSUE TITLE
script to update all included modules

### ISSUE THREAD

**Issue by j-schwartz** [rachelproject/contentshell#8]
it'd be great to have a script that first read which modules are in /var/www/modules and then, when ran, updated those modules against the master database

**Comment by spatiald** [comment]
@j-schwartz You reference /var/www/modules...the enhancement needs to be cross-compatible and should check what type of device it is.  It might be a good idea for us to add a file to all the hardware we modify...this way, the contentshell scripts can parse that to determine where the appropriate files would be.

/etc/rachel-release (on a CAP)
```
NAME="RACHEL-Plus"
VERSION="1.2.16"
ID=rachel
ID_LIKE=cap
PRETTY_NAME="RACHEL-Plus (1.2.16)"
VERSION_ID="1.2.16"
```

/etc/rachel-release (on a Pi)
```
NAME="RACHEL-Pi"
VERSION="<replace with applicable version of Ubuntu?>"
ID=rachel
ID_LIKE=pi
PRETTY_NAME="RACHEL-Pi (<Ubuntu verison?>)"
VERSION_ID="<Ubuntu version?>"
```

Also, I have update code in the RACHEL configure script for the CAP...here is the function:
```
contentModuleInstall(){
    trap ctrlC INT
    if [[ -f /tmp/module.lst ]]; then
        echo; printStatus "Your previously selected module list:"
        # Sort/unique the module list
        cat /tmp/module.lst
        echo; printQuestion "Do you want to use this module list?"
        read -p "    Enter (Y/n) " REPLY
        if [[ $REPLY =~ ^[Nn]$ ]]; then rm -f /tmp/module.lst; fi
    fi
    SELECTMODULE=1
    MODULELIST=$(rsync --list-only $RSYNCDIR/rachelmods | egrep '^d' | awk '{print $5}' | tail -n +2)
    while [[ $SELECTMODULE == 1 ]]; do
        echo; printStatus "What RACHEL module would you like to select for download or update?"
        echo "(Ctrl-C to cancel module install)"
        echo
        select module in $MODULELIST; do 
            echo "$module" >> /tmp/module.lst
            echo; printStatus "Added module $module to the install/update cue."
            break
        done
        echo; printStatus "Your selected module list:"
        sort -u /tmp/module.lst > /tmp/module.tmp; mv /tmp/module.tmp /tmp/module.lst
        cat /tmp/module.lst
        echo; printQuestion "Do you want to select another module?"
        read -p "    Enter (y/N) " REPLY
        if [[ $REPLY =~ ^[Yy]$ ]]; then SELECTMODULE=1; else SELECTMODULE=0; fi
    done
    echo; printQuestion "Are you ready to install your selected modules?"
    read -p "    Enter (y/N) " REPLY
    if [[ $REPLY =~ ^[Yy]$ ]]; then
        while read m; do
            echo; printStatus "Downloading $m"
            rsync -avz $RSYNCDIR/rachelmods/$m $RACHELWWW/modules/
            commandStatus
            printGood "Done."
        done < /tmp/module.lst
    fi
    rm -f /tmp/module.lst
}
```

**Comment by needlestack** [comment]
Well we're not there yet - but I did just add the ability to update contentshell itself, which is a start. Going forward people can go to the "version" page, check if there's a new contentshell, and update if there is.

I've done some work done on module updating, but it's not ready for prime time yet.

Also: I've decided to go with semantic tags (v2.1.0, for example) and I've cleaned up the tags/releases on contentshell, rachelpios, and module-template -- @spatiald, do you want to do the same with rachelplus?

**Comment by spatiald** [comment]
Yup, semantic tags are fine with me.

On Oct 23, 2016, at 19:43, Jonathan Field <notifications@github.com> wrote:

Well we're not there yet - but I did just add the ability to update
contentshell itself, which is a start. Going forward people can go to the
"version" page, check if there's a new contentshell, and update if there is.

I've done some work done on module updating, but it's not ready for prime
time yet.

Also: I've decided to go with semantic tags (v2.1.0, for example) and I've
cleaned up the tags/releases on contentshell, rachelpios, and
module-template -- @spatiald <https://github.com/spatiald>, do you want to
do the same with rachelplus?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/rachelproject/contentshell/issues/8#issuecomment-255623126>,
or mute the thread

**Comment by needlestack** [comment]
Ok, this is released as v2.2.0. Updates are under the "version" tab, and you can install and delete modules under the "install" tab. Let me know what you experience!
